### DISCUSSION TITLE
Can I include a Struct structure inside a Union Topic Type?

### DISCUSSION THREAD

**Discussion by rla0h** [OpenDDS/OpenDDS#4789] [Q&A]
Hi, I'm studying Union Topic Type in C++,
I used IDL that struct structure inside a union topic type. 
Like this..
```IDL
module Test {
    struct Message {
        string from;
        string subject;
        string text;
        long count;
    };
    enum TypeKind {
        STRING_TYPE,
        LONG_TYPE,
        STRUCT_TYPE
    };
    @topic
    union UnionTypeTest switch(@key TypeKind) {
    case STRING_TYPE:
        string string_data;
    case LONG_TYPE:
        long long_data;
    case STRUCT_TYPE:
        Message m;
    };
};
```

OpenDDS Build was successed, but when I published Message Data, it occured Core dump.

so, Can I include a Struct structure inside a Union Topic Type?

**Comment by jwillemsen** [comment]
Yes, that is possible, very likely you have a bug in your code, see https://github.com/OpenDDS/OpenDDS/tree/master/tests/DCPS/UnionTopic for a unit test where you can borrow code from.

Maybe run your code with valgrind or use gdb to check the core.

**Comment by jwillemsen** [comment]
Hi,

Yes, that is possible, very likely you have a bug in your code, see 
https://github.com/OpenDDS/OpenDDS/tree/master/tests/DCPS/UnionTopic for 
a unit test where you can borrow code from.

Maybe run your code with valgrind or use gdb to check the core.

Regards,

Johnny

**Reply by rla0h** [reply]
Thank you for your reply.
Is the Union Type Test you provided working properly?
I am currently reinstalling OpenDDS because the Union Type Test is showing Failed.

**Reply by rla0h** [reply]
I had reinstalled OpenDDS, but It has error
```bash
ERROR: executable "UnionTopic" does not exist; subdir: ./; basename: UnionTopic ; dirname: .
ERROR: no process with name=test
test FAILED.
```

**Reply by mitza-oci** [reply]
The test is passing in automated builds https://scoreboard.opendds.org/dds.matrix.html
On your local setup, the test program was not built.
